src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -21,40 +27,65 @@ export interface DialogProps extends ModalProps {
     acceptButton: ButtonProps;
     /** Properties of the cancel button */
     cancelButton?: ButtonProps;
+    /** the context of the buttons in the footer */
+    buttonContext?: Omit<ButtonContext, 'secondary'>;
+    /** closes the dialog */
+    onClose?: () => void;
 }
 
 const { block, elem } = bem('Dialog', styles);
 
 export const Dialog: React.FC<DialogProps> = (props) => {
-    const { isOpen, children, title, acceptButton, cancelButton, ...rest } = props;
-
+    const {
+        onDialogTrigger,
+        isOpen,
+        children,
+        title,
+        acceptButton,
+        cancelButton,
+        dialogTrigger,
+        onClose,
+        ...rest
+    } = props;
     const isConfirm = !!cancelButton;
 
     return (
-        <Modal {...rest} {...block(props)} isOpen={isOpen}>
-            <div {...elem('content', props)} role="alert">
-                {!!title && (
-                    <Heading level="h2" {...elem('title', props)}>
-                        {title}
-                    </Heading>
-                )}
-                {children}
-            </div>
-            <div {...elem('actions', props)}>
-                {isConfirm && (
+        <div {...block(props)}>
+            {dialogTrigger && (
```

Comment:
this button is not needed. Also we would ended up having a Button inside of a button.